packages/components/src/presentation-components/properties/inputs/UniquePropertyValuesSelector.tsx

```diff
@@ -221,27 +198,46 @@ function getFieldClassInfos(field?: Field) {
 
 interface UseUniquePropertyValuesLoaderProps {
   imodel: IModelConnection;
+  property: PropertyDescription;
+  descriptor: Descriptor;
   ruleset?: Ruleset;
   field?: Field;
   descriptorInputKeys?: Keys;
 }
 
-function useUniquePropertyValuesLoader({ imodel, ruleset, field, descriptorInputKeys }: UseUniquePropertyValuesLoaderProps) {
+function useUniquePropertyValuesLoader({ imodel, property, descriptor, ruleset, field, descriptorInputKeys }: UseUniquePropertyValuesLoaderProps) {
+  const [loadedCount, setLoadedCount] = useState(0);
+  const [loadedOptions, setLoadedOptions] = useState<UniqueValue[]>([]);
+  const [hasMore, setHasMore] = useState(false);
+
+  useEffect(() => {
+    setLoadedOptions([]);
```

Comment:
Shouldn't we reset `loadedCount` too?

Reply:
We should, fixed and updated tests to verify that the correct page index is used.